CMakeLists.txt

```diff
@@ -12,6 +12,9 @@ option(S2N_NO_PQ "Disables all Post Quantum Crypto code. You likely want this
 for older compilers or uncommon platforms." OFF)
 option(S2N_NO_PQ_ASM "Turns off the ASM for PQ Crypto even if it's available for the toolchain.
 You likely want this on older compilers." OFF)
+option(VISIBILITY_PUBLIC "Private internal symbols will be exported. This is enabled by default when building with -DBUILD_SHARED_LIBS=ON (default=OFF) and -DBUILD_TESTING=ON (default=ON)" OFF)
```

Comment:
Do we actually need this option? Are any of the current use cases not handled by checking for BUILD_TESTING?

Reply:
For current use cases, I don't think we need it. It would allow more flexibility in the future, but I'm fine removing it

Reply:
@camshaft, you had suggested making it explicit, do you still think we should include it?

Reply:
The only use case we know of right now is running tests. It's probably fine to drop it for now and revisit if someone has an additional need.

Reply:
This setting only impacts the linker, so has no impact on a static lib.  Therefore linking the tests with the static lib would mean this option isn't required, which is what we do for the Makefile version.

Exporting symbols while building dynamically is the default behavior of gcc and clang.